tuna/miopen/worker/fin_class.py

```diff
@@ -309,15 +309,30 @@ def __set_all_configs(self, idx=0, num_blk=1):
         query = query.order_by(self.dbt.config_table.id)
         rows = query.all()
 
+        len_rows = len(rows)
         master_cfg_list = []
         for row in rows:
           r_dict = compose_config_obj(row, self.config_type)
           if self.config_type == ConfigType.batch_norm:
             r_dict['direction'] = row.get_direction()
           master_cfg_list.append(r_dict)
 
-        block_size = len(rows) // num_blk  #size of the config block
-        extra = len(rows) % num_blk  #leftover configs, don't divide evenly
+        #remove old applicability
+        rm_old = ''
+        if self.label:
+          cfg_ids = [str(row.id) for row in rows]
+          cfg_str = ','.join(cfg_ids)
+          rm_old = f"update {self.dbt.solver_app.__tablename__} set applicable=0 where session={self.session_id} and config in ({cfg_str});"
```

Comment:
we never actually delete from the applicability table? we just set it to not applicable? this will make this table very very big. Is there a reason to keep old applicability around?

Reply:
I think setting the applicability field was faster than attempting to delete the entry. Anyhow, this only updates the one session where we are creating the present applicability set. I suppose it would be good to remove old sessions and with it data tied to those sessions like applicability, once those tuning sessions are no longer necessary.